### PR TITLE
CAT-1333 Update PMD to 5.3+

### PR DESCRIPTION
**Pull request by M-Schiller**

### CODE REVIEW

File: catroid/src/androidTest/java/org/catrobat/catroid/uitest/content/brick/PhiroSeekBarColorBrickTest.java [hunk 1]
```diff
@@ -96,7 +98,7 @@ private void testFormulaEditorInput(int phiroLedColorEditText, int phiroLedColor
 		solo.clickOnView(solo.getView(phiroLedColorEditText));
 		solo.clickOnView(solo.getView(phiroLedColorTextView));
 
-		UiTestUtils.insertDoubleIntoEditText(solo, 100.0);
+		insertDoubleIntoEditText(solo, 100.0);
```
Comment (robertpainsi) [inline catroid/src/androidTest/java/org/catrobat/catroid/uitest/content/brick/PhiroSeekBarColorBrickTest.java]: Hmm, did pmd really complain about this? If I undo your changes here, I still don't get any pmd warnings.
If this isn't something pmd complains, please undo :smirk_cat:

**Comment by robertpainsi** [comment]
> Applying rule set filter for exclusions: The rule "BooleanInversion" in ruleset "controversial" has been removed from PMD and no longer exists. Please change your ruleset!

**Comment by 84n4n4** [comment]
test this please

**Comment by M-Schiller** [comment]
@robertpainsi Can you think of an alternative to UiTestUtils.java:421 since this is the only place in code that requires  excluding the rule  `AvoidDecimalLiteralsInBigDecimalConstructor`?

**Comment by robertpainsi** [comment]
@M-Schiller, not really. Using the _BigDecimal_ constructor seems to be the easiest way to convert a double to a string without an exponent. Since it's used in tests only, it's fine to just [suppress pmd](http://pmd.sourceforge.net/pmd-4.3/suppressing.html) at this line only.

**Comment by M-Schiller** [comment]
I think all changes unrelated to pmd should be excluded from this commit. I think UiTestUtils should be consistent regarding its public static methods used inside the class itself:
should they _all_ be called like this:
`UiTestUtils.myPublicStaticMethod(...)`
or like this:
`myPublicStaticMethod(...)`?

**Comment by robertpainsi** [comment]
Great! Looking good!
ok to test

> consistent regarding its public static methods used inside the class itself:
should they all be called like this:
`UiTestUtils.myPublicStaticMethod(...)`
or like this:
`myPublicStaticMethod(...)`?

Is there a PMD or Checkstyle rule to check for that?

**Comment by robertpainsi** [comment]
please run espresso tests

**Comment by M-Schiller** [comment]
No, at least not yet.

**Comment by robertpainsi** [comment]
In general, I personally prefer the second one because the class reference is redundant.
The class `uitest.util.UiTestUtils` will be removed once we moved all robotium UI tests to espresso.
IMO it's not necessary to go throw other files and remove redundant class references.

If you want to, you can create a ticket on Checkstyle for such check.

**Pull request merged**